src/cryptography/hazmat/primitives/twofactor/utils.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import unicode_literals
+
+import base64
+
+from six.moves.urllib.parse import quote, urlencode
+
+
+__all__ = ['get_provisioning_uri']
+
+
+def get_provisioning_uri(otp, account_name, issuer=None, counter=None):
+    """Generates a provisioning URI which can be recognized by Two-Factor
+    Authentication Apps. See also: http://git.io/vkvvY
+
+    :param otp: An instance of
+                :class:`cryptography.hazmat.primitives.twofactor.hotp.HOTP` or
+                :class:`cryptography.hazmat.primitives.twofactor.totp.TOTP`.
+    :param account_name: The display name of account, such as
+                         ``'Alice Smith'`` or ``'alice@example.com'``.
+    :param issuer: The display name of issuer.
+    :param counter: The current value of counter. It is required for HOTP.
+    :return: The URI string.
+    :raises RuntimeError: if counter is missing but otp type is HOTP
+    """
+    hotp = getattr(otp, '_hotp', otp)
```

Comment:
If I'm reading this right you are attempting to determine if the otp object is an instance of HOTP and not TOTP? If so this should be rewritten as `isinstance(otp, HOTP)` instead.